stats/src/main/java/io/airlift/stats/CpuTimer.java

```diff
@@ -66,24 +81,28 @@ public CpuDuration elapsedIntervalTime()
     {
         long currentWallTime = System.nanoTime();
         long currentCpuTime = THREAD_MX_BEAN.getCurrentThreadCpuTime();
-        long currentUserTime = THREAD_MX_BEAN.getCurrentThreadUserTime();
+        // ThreadMXBean will return -1 if user CPU time collection is not supported
+        long currentUserTime = (collectUserTime && intervalUserStart != -1) ? THREAD_MX_BEAN.getCurrentThreadUserTime() : -1;
 
         return new CpuDuration(
                 nanosBetween(intervalWallStart, currentWallTime),
                 nanosBetween(intervalCpuStart, currentCpuTime),
-                nanosBetween(intervalUserStart, currentUserTime));
+                // currentUserTime is -1 when ThreadMXBean does not support user collection or when collectUserTime is false
+                currentUserTime == -1 ? null : nanosBetween(intervalUserStart, currentUserTime));
     }
 
     public CpuDuration elapsedTime()
     {
         long currentWallTime = System.nanoTime();
         long currentCpuTime = THREAD_MX_BEAN.getCurrentThreadCpuTime();
-        long currentUserTime = THREAD_MX_BEAN.getCurrentThreadUserTime();
+        // ThreadMXBean will return -1 if user CPU time collection is not supported
+        long currentUserTime = (collectUserTime && userStartTime != -1) ? THREAD_MX_BEAN.getCurrentThreadUserTime() : -1;
```

Comment:
No need to check `userStartTime != -1` ?

Reply:
When `collectUserTime == true` but `ThreadMXBean.getCurrentThreadUserTime()` has already returned `-1` (not supported) on the initial attempt, we know that we can't produce a user time metric anyway and will skip the call. If for whatever reason the return value is not stable, we don't want to attempt to subtract an error code from whatever this call returns to produce a user time metric anyway.

Reply:
I've decided this can be simplified to remove the `collectUserTime` boolean and to consolidate the logic to only need to test whether the user start time is `-1`.